### ISSUE TITLE
Error assembling the diffraction data

### ISSUE THREAD

**Issue by marineanalytica** [mancellin/capytaine#37]
Hi, 
When assembling the diffraction data into a labelled array the version 1.2 gives the following error:

 ```
File "pandas\_libs\index.pyx", line 111, in pandas._libs.index.IndexEngine.get_loc

  File "pandas\_libs\index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc

  File "pandas\_libs\hashtable_class_helper.pxi", line 1619, in pandas._libs.hashtable.PyObjectHashTable.get_item

  File "pandas\_libs\hashtable_class_helper.pxi", line 1627, in pandas._libs.hashtable.PyObjectHashTable.get_item

KeyError: 'radiating_dof'
```

Do you have the same issue? Any clue what the problem can be?
Thanks,

**Comment by marineanalytica** [comment]
It happens running the command:

diffraction_data = cpt.assemble_dataset(diffraction_results)

**Comment by mancellin** [comment]
It might be a bug.

I'm guessing `diffraction_results` is a list of only `DiffractionResult` objects without any `RadiationResult`. Is that right? Is everything fine if you had a single `RadiationResult` to the list?

**Comment by marineanalytica** [comment]
yes, you're right. If I put both radiation and diffraction data in the same list the problem is gone.

**Comment by marineanalytica** [comment]
Hey btw thanks for sharing the code. It's just great!

**Comment by mancellin** [comment]
Thank you for the kind words.

Actually, I'll let the issue open, for me to remember to fix the all-diffraction case whenever I have some time.

**Comment by ryancoe** [comment]
fixed via 31cc4c9?

**Comment by mancellin** [comment]
Yes
